application/src/main/java/uk/nhs/hee/tis/revalidation/service/GmcDoctorConnectionSyncService.java

```diff
@@ -64,11 +68,25 @@ public void receiveMessage(final String gmcSyncStart) {
 
   private void sendToSqsQueue(final List<DoctorsForDB> gmcDoctors) {
     gmcDoctors.stream()
-        .forEach(doctor -> queueMessagingTemplate.convertAndSend(sqsEndPoint, doctor));
+        .forEach(doctor ->
+            {
+              final var summary = RevalidationSummaryDto.builder()
+                  .doctor(doctor)
+                  .gmcOutcome(getGmcOutcomeForDoctor(doctor.getGmcReferenceNumber()))
+                  .syncEnd(false)
+                  .build();
+              queueMessagingTemplate.convertAndSend(sqsEndPoint, summary);
+            }
+        );
+
     log.info("GMC doctors have been published to the SQS queue ");
 
-    DoctorsForDB syncEnd = DoctorsForDB.builder().syncEnd(true).build();
+    RevalidationSummaryDto syncEnd = RevalidationSummaryDto.builder().syncEnd(true).build();
```

Comment:
AFAIK, the default for SQS is an unordered queue.  If we pick FIFO, we'll need to make sure there are some really robust compensatory action mechanisms for failed messages.

I was going to say that sending a "sync end" flag can at most be an "finished requesting" signal but it is feasible (just more to do) to work out what to do with a retry that is processed after a "syncEnd" for the same 'saga'. <- I haven't looked at the saga pattern for a couple of months

Reply:
Agree this is important, I think this will have to be done on the consumer service end though